File: Easy/JavaScript/414. Third Maximum Number.js
var thirdMax = function(nums) {
    nums.sort((a, b) => a - b);
    let count = 0;
    let third = nums[0];

    for(let i = nums.length - 1; 0 <= i; i--){
        if(third != nums[i]){
            third = nums[i];
            count++;
        }
        if(count == 3)
            break;
    }

    return count == 3 ? third : nums[nums.length - 1];
};

File: Easy/Java/414. Third Maximum Number.java
class Solution {
    public int thirdMax(int[] nums) {
        Arrays.sort(nums);
        int count = 0;
        int third = nums[0];

        for(int i = nums.length - 1; 0 <= i; i--){
            if(third != nums[i]){
                third = nums[i];
                count++;
            }
            if(count == 3)
                break;
        }

        return count == 3 ? third : nums[nums.length - 1];
    }
}
